### PR TITLE
push & delete: support v3 file system

### PR DESCRIPTION
**Pull request by yugangw-msft**
I moved the "OfflineFeedUtility" to nuget.core, which contains the core logic of pushing to v3 file system.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs [hunk 1]
```diff
@@ -563,5 +607,37 @@ public SearchPathResult(string path, bool isFile)
...
+                    // ~/Foo/Foo.1.0.0.nuspec (LocalPackageRepository with PackageSaveMode.Nuspec)
+                    return true;
+                }
+                var idDirectoryName = Path.GetFileName(idDirectory);
+                foreach (var versionDirectoryPath in Directory.GetDirectories(idDirectory))
+                {
+                    if (Directory.GetFiles(versionDirectoryPath, idDirectoryName + NuGetConstants.ManifestExtension).Any())
+                    {
+                        // If we have files in the format {packageId}/{version}/{packageId}.nuspec, assume it's an expanded package repository.
+                        return false;
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs:635]: Why is this check needed?

**Comment by emgarten** [comment]
Is it possible to push a package to a new v3 folder? It looks like this needs an additional flag, or maybe it should default to v3 for new folders.

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs (hunk 1 above)
Comment (yugangw-msft) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/PackageUpdateResource.cs:635]: i ported from nuget2 strictly, for the sole backward compatibility reason. I don't have too much context in this area.

**Comment by yugangw-msft** [comment]
I am fine of going default to V3, if it offers benefits which appear so

**Comment by yugangw-msft** [comment]
Closing this one, please uses #280 which triggers CI for easier cross check.

**Pull request closed without merging**